src/Sylius/Component/Order/Context/ResettingCartContextInterface.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Order\Context;
+
+/** @deprecated since 1.13 and will be removed in Sylius 2.0. */
+interface ResettingCartContextInterface
```

Comment:
Couldn't we just use the `Symfony\Contracts\Service\ResetInterface` instead of creating a new, immediately deprecated one?

Reply:
I would go as far to extend the ResetInterface, but i honestly don't see the benefit of using `ResetInterface` over a bespoke one.

```

class X {
    public function __construct(
        \Symfony\Contracts\Service\ResetInterface $restableCartButUpToMeWhatToNameThis,
    )
    {

    }
}
```
vs

```
class X {
    public function __construct(
        Sylius\Component\Order\Context\ResettingCartContextInterface $canGiveThisVarAnyName,
    )
    {

    }
}

```

Reply:
Not arguing though, happy to make the change. Leave a thumbs down and I'll proceed with your suggestion

Reply:
Putting aside the fact that we're introducing a new interface that is immediately deprecated and presumably won't exist in the next major.
My main issue with creating this new interface is that the intent is the same: this service is stateful and should be reset. That's precisely what the `ResetInterface` is for, there's even [autoconfiguration](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L628) set on the framework extension that auto tags any service with this interface with `kernel.reset`.
IMO the Symfony logic in this case fits to a tee.